oauth2_http/java/com/google/auth/oauth2/TrustBoundary.java

```diff
@@ -0,0 +1,227 @@
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("encodedLocations", encodedLocations)
+          .add("locations", locations)
+          .toString();
+    }
+  }
+
+  @VisibleForTesting
+  static void setEnvironmentProviderForTest(@Nullable EnvironmentProvider provider) {
+    environmentProvider = provider == null ? SystemEnvironmentProvider.getInstance() : provider;
+  }
+
+  /**
+   * Checks if the trust boundary feature is enabled based on an environment variable. The feature
+   * is enabled if the environment variable is set to "true" or "1" (case-insensitive). Any other
+   * value, or if the variable is unset, will result in the feature being disabled.
+   *
+   * @return True if the trust boundary feature is enabled, false otherwise.
+   */
+  static boolean isTrustBoundaryEnabled() {
+    String trustBoundaryEnabled =
+        environmentProvider.getEnv(GOOGLE_AUTH_TRUST_BOUNDARY_ENABLED_ENV_VAR);
+    if (trustBoundaryEnabled == null) {
+      return false;
+    }
+    String lowercasedTrustBoundaryEnabled = trustBoundaryEnabled.toLowerCase();
+    return "true".equals(lowercasedTrustBoundaryEnabled) || "1".equals(trustBoundaryEnabled);
+  }
+
+  /**
+   * Refreshes the trust boundary by making a network call to the trust boundary endpoint.
+   *
+   * @param transportFactory The HTTP transport factory to use for the network request.
+   * @param url The URL of the trust boundary endpoint.
+   * @param accessToken The access token to authenticate the request.
+   * @param cachedTrustBoundary An optional previously cached trust boundary, which may be used in
+   *     the request headers.
+   * @return A new TrustBoundary object containing the refreshed information.
+   * @throws IllegalArgumentException If the provided access token is null or expired.
+   * @throws IOException If a network error occurs or the response is malformed.
+   */
+  static TrustBoundary refresh(
+      HttpTransportFactory transportFactory,
+      String url,
+      AccessToken accessToken,
+      @Nullable TrustBoundary cachedTrustBoundary)
+      throws IOException {
+    Preconditions.checkNotNull(accessToken, "The provided access token is null.");
+    if (accessToken.getExpirationTime() != null
+        && accessToken.getExpirationTime().before(new Date())) {
+      throw new IllegalArgumentException("The provided access token is expired.");
+    }
+
+    HttpRequestFactory requestFactory = transportFactory.create().createRequestFactory();
+    HttpRequest request = requestFactory.buildGetRequest(new GenericUrl(url));
+    request.getHeaders().setAuthorization("Bearer " + accessToken.getTokenValue());
+
+    // Add the cached trust boundary header, if available.
+    if (cachedTrustBoundary != null) {
+      String headerValue =
+          cachedTrustBoundary.isNoOp() ? "" : cachedTrustBoundary.getEncodedLocations();
+      request.getHeaders().set(TRUST_BOUNDARY_KEY, headerValue);
+    }
```

Comment:
I think I missed this earlier. Why does the call to `/allowedLocations` also need the header? I understand that the header needs to be added to client library calls. I might have missed this part in the doc